### PR TITLE
IGNITE-16670

### PR DESCRIPTION
**Pull request by denis-chudov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/manager/RECOVERY.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+### Definitions
+
+**Local state** is a state of components, started tables, schemas, persistent and in-memory storages, partitions, RAFT groups, etc. on a given node. It also includes a projection of distributed cluster configuration.
+
+**Local state recovery** is changing local state in a way that every component on the node has its state configured and updated accordingly to cluster-wide configuration, topology, stored data, etc.
+
+**Vault** is a local storage of node configuration and a projection of cluster configuration. Vault has no log of historical changes. See [Vault](../../../../../../../../../vault/README.md).
+
+**Metastorage** is a distributed storage of cluster-wide meta information, including cluster [configuration](../../../../../../../../../configuration/README.md). Metastorage is located on a group of nodes, forming a RAFT group. It stores a history of changes.
```
Comment (sk0x50) [inline modules/core/src/main/java/org/apache/ignite/internal/manager/RECOVERY.md:11]:
```suggestion
**Meta storage** is a distributed storage of cluster-wide meta information, including cluster [configuration](../../../../../../../../../configuration/README.md). Meta storage is located on a group of nodes, forming a RAFT group. It stores a history of changes.
```

File: modules/core/src/main/java/org/apache/ignite/internal/manager/RECOVERY.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+**Logical topology** - a set of nodes that are present in physical topology and finished recovery process. They are considered to be able to serve user load.
+
+### Prerequisites
+
+1. Vault is either empty or contains data that is available for reading (no corruption).
+   If any action is required to recover data in Vault, it should be executed before.
+2. Data in Vault is consistent to the moment when the node was stopped (last applied revision appropriates to the rest of the data).
+3. Vault contains local node configuration, and a projection of cluster configuration. The latter can be written only after the cluster is initialized. No conflicts are possible between the node local configuration and the cluster configuration that is stored in Metastorage because the properties’ sets don’t intersect with each other.
+4. The prerequisite for the recovery from Metastorage is receiving a successful _nodeJoinResponse_, see [Node join protocol](TODO add readme).
```
Comment (kgusakov) [inline modules/core/src/main/java/org/apache/ignite/internal/manager/RECOVERY.md:27]: Do we have an issue for this TODO? As I know, we can't merge any TODOs without issue number.

File: modules/core/src/main/java/org/apache/ignite/internal/manager/RECOVERY.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+_b._ **LM > LV and MM <= LV + 1**: this is the case when Vault is behind the cluster but the node is able to get all sequential updates from Metastorage and apply them to components. The node can recover components’ state using data from Vault and then, when all components have started, process further updates from Metastorage log as Metastorage notifications, in historical sequence, using listeners. This works as if each component listener received each configuration update from cluster one by one and handled it as a single change. The listeners start handling notifications after all components have started.
+  
+During this process new updates can be written to Metastorage. After applying updates, the node should check the current value of LM and if it is significantly greater than LV, it should continue receiving updates from Metastorage, trying to catch current LM. This process ends when the difference between LV and LM is less or equal to some **acceptable difference value**, meaning that now the difference between LM and LV is small enough to let the node join logical topology.
```
Comment (kgusakov) [inline modules/core/src/main/java/org/apache/ignite/internal/manager/RECOVERY.md:76]: How do we can configure "acceptable difference value"?

File: modules/README.md [hunk 4]
```diff
@@ -16,6 +16,8 @@ Module Name | Description
 [cli-common](cli-common/README.md)|Shared interfaces definitions for pluggable CLIng
 [configuration](configuration/README.md)|Ignite configuration classes and configuration management framework
 [configuration-annotation-processor](configuration-annotation-processor/README.md)|Tooling for generating Ignite configuration model classes from configuration schema definition
+[core: causality tokens](core/src/main/java/org/apache/ignite/internal/causality/README.md)|Causality tokens
```
[the remark this answers is not in the document]
Comment (denis-chudov) [reply modules/README.md:19]: I just followed a template that already exists, when README is located inside of the package, like modules/api/src/main/java/org/apache/ignite/table/mapper/README.md or modules/schema/src/main/java/org/apache/ignite/internal/schema/README.md.

**Pull request closed without merging**